tutorials/physics/using_jolt_physics.rst

```diff
@@ -0,0 +1,473 @@
+
+The "recovery" step in Godot Physics is hardcoded to always do 4 iterations with 40%
+depenetration per iteration. I figured these constants might be useful to expose, so
+when using this module they can be configured in the project settings as
+:ref:`physics/jolt_physics_3d/motion_queries/recovery_iterations<class_ProjectSettings_Property_physics/jolt_physics_3d/motion_queries/recovery_iterations>`
+and :ref:`physics/jolt_physics_3d/motion_queries/recovery_amount<class_ProjectSettings_Property_physics/jolt_physics_3d/motion_queries/recovery_amount>`
+respectively.
+
+The implementation of :ref:`PhysicsServer3D.body_test_motion()<class_PhysicsServer3D_method_body_test_motion>`
+in this module also differs slightly from the Godot Physics implementation, as
+replicating the Godot Physics version resulted in a prohibitive amount of ghost
+collisions for :ref:`move_and_slide<class_CharacterBody3D_method_move_and_slide>`.
+
+The discrepancies are as follows:
+
+- There is no TEST_MOTION_MIN_CONTACT_DEPTH_FACTOR.
+- There is no epsilon added to the safe margin during recovery.
+- Contacts with normals that are not opposing the motion vector are discarded from
+  the final collision check.
+
+While these discrepancies do seem to result in less ghost collisions when using
+Jolt, they also introduce new problems:
+
+- :ref:`move_and_slide<class_CharacterBody3D_method_move_and_slide>` is generally
+  slower than in Godot Physics, due to :ref:`class_CharacterBody3D`::_snap_on_floor(PR NOTE: Apply_floor_snap?)
+  triggering on every call.
+- :ref:`class_CharacterBody3D` will not report wall collisions unless you're moving towards them.
+
+It's not clear to me how to fix these issues, but I suspect they will require
+changes to move_and_slide on the scene side of things.
+
+Ghost collisions
+~~~~~~~~~~~~~~~~
+
+Jolt employs two techniques to mitigate ghost collisions, meaning collisions with
+internal edges of shapes/bodies.
+
+The first technique, called "active edge detection", marks edges of triangles in
+:ref:`class_ConcavePolygonShape3D` or :ref:`class_HeightMapShape3D` as either "active" or "inactive", based on
+the angle to the neighboring triangle. When a collision happens with an inactive
+edge the collision normal will be replaced with the triangle's normal instead, to
+lessen the effect of ghost collisions.
+
+The angle threshold for this active edge detection is configurable through the
+project setting :ref:`physics/jolt_physics_3d/collisions/active_edge_threshold<class_ProjectSettings_property_physics/jolt_physics_3d/collisions/active_edge_threshold>`.
+
+The second technique, called "enhanced internal edge removal", instead adds runtime
+checks to detect whether an edge is active or inactive, based on the contact points
+of the two bodies. This has the benefit of applying not only to collisions with
+:ref:`class_ConcavePolygonShape3D` and :ref:`class_HeightMapShape3D`, but also edges between any shapes within
+the same body.
```

Comment:
This is an important point and good to stress: If you want to avoid ghost collisions, make sure to group multiple shapes within the same body. It does not work if an object is sliding over the boundary between two static bodies.